markdown/plext/using_plpgsql.html.md.erb

```diff
@@ -19,143 +19,278 @@ software distributed under the License is distributed on an
 
-PL/pgSQL functions can also be declared with output parameters in place of an explicit specification of the return type. This does not add any fundamental capability to the language, but it is often convenient, especially for returning multiple values. The RETURNS TABLE notation can also be used in place of RETURNS SETOF .
+## <a id="supportedargumentandresultdatatypes"></a>PL/pgSQL Argument and Result Data Types 
 
-This topic describes the following PL/pgSQLconcepts:
+Functions written in PL/pgSQL accept as arguments any base or array data type supported by the server, and they can return a result containing any of these data types. PL/pgSQL functions can also accept and return any composite type (row-type) specified by name.
 
--   [Table Functions](#tablefunctions)
--   [SQL Functions with Variable number of Arguments](#sqlfunctionswithvariablenumbersofarguments)
--   [Polymorphic Types](#polymorphictypes)
+You can declare PL/pgSQL functions to accept and return the polymorphic types `anyelement` and `anyarray` types. PL/pgSQL functions can also be declared to return a set (or table) of any data type that can be returned as a single instance. Finally, you can declare a PL/pgSQL function to return `void` if it has no useful return value.
 
+In place of an explicit specification of the return type, you can declare PL/pgSQL functions with output parameters. This does not add any fundamental capability to the language, but it is often convenient, especially when returning multiple values.
 
-## <a id="tablefunctions"></a>Table Functions 
+Upcoming sections provide specific PL/pgSQL examples using base, composite, and polymorphic argument and return types.
 
 
-Table functions are functions that produce a set of rows, made up of either base data types (scalar types) or composite data types (table rows). They are used like a table, view, or subquery in the FROM clause of a query. Columns returned by table functions can be included in SELECT, JOIN, or WHERE clauses in the same manner as a table, view, or subquery column.
+### <a id="plpgsql_namingargs"></a>Naming PL/pgSQL Function Arguments
 
-If a table function returns a base data type, the single result column name matches the function name. If the function returns a composite type, the result columns get the same names as the individual attributes of the type.
+Arguments passed to PL/pgSQL functions are named with identfiers `$1`, `$2`, `$3`, etc. If you chose, you can also declare aliases for the `$<n>` argument names.
 
-A table function can be aliased in the FROM clause, but it also can be left unaliased. If a function is used in the FROM clause with no alias, the function name is used as the resulting table name.
+One way to declare an alias is to give the argument a name in the PL/pgSQL function signature. In the following example, the single input argument (`$1`) is named `subtotal`. `subtotal` is used by name in the sales tax calculation in the body of the function.
```

Comment:
Maybe remove parens from (`$1`)  Seems a little cumbersome when you're dealing with specific syntax like this.